src/h3/geo.pyx

```diff
@@ -50,58 +73,100 @@ cdef h3lib.Geofence make_geofence(geos):
     return gf
 
 
+cdef free_geofence(h3lib.Geofence* gf):
+    if gf.verts:
+        stdlib.free(gf.verts)
+    gf.verts = NULL
+
+
 cdef class GeoPolygon:
     """ Basic version of GeoPolygon
-    Doesn't work with holes.
+
     """
     cdef:
         h3lib.GeoPolygon gp
 
-    def __cinit__(self, geos):
-        self.gp.numHoles = 0
-        self.gp.holes = NULL
-        self.gp.geofence = make_geofence(geos)
+    def __cinit__(self, outer, holes=None):
+        """
 
-    def __dealloc__(self):
-        if self.gp.geofence.verts:
-            stdlib.free(self.gp.geofence.verts)
-        self.gp.geofence.verts = NULL
-
-
-
-cpdef h3lib.H3int geo_to_h3(double lat, double lng, int res) except 1:
-    cdef:
-        h3lib.GeoCoord c
+        Parameters
+        ----------
+        outer : list or tuple
+            GeoFence
+            A GeoFence is a sequence of >= 4 (lat, lng) pairs where the last
+            element is the same as the first.
```

Comment:
I don't believe the C lib cares whether the loop is closed or not.

Reply:
Oooh! Good to know. I'll try that out and add something about that to the docstrings.

@nrabinowitz, is that documented in the C lib? I could do a pull request. Would it belong in `h3api.h.in`?

Reply:
I added a test to confirm that this is true. It takes a polygon with 2 holes and checks every permutation of each LinearRing being clockwise or counterclockwise and being or not being a closed loop.